notarization_poller/src/notarization_poller/config.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+"""Config for scriptworker.
+
+Attributes:
+    DEFAULT_CONFIG (frozendict): the default configuration
+    log (logging.Logger): the log object for the module.
+
+"""
+import argparse
+import logging
+import logging.handlers
+import os
+from copy import deepcopy
+
+from frozendict import frozendict
+from yaml import safe_load
+
+from notarization_poller.constants import DEFAULT_CONFIG
+from notarization_poller.exceptions import ConfigError
+
+log = logging.getLogger(__name__)
+
+
+def update_logging_config(config, log_name=None, file_name="worker.log"):
+    """Update python logging settings from config.
+
+    By default, this sets the ``scriptworker`` log settings, but this will
+    change if some other package calls this function or specifies the ``log_name``.
+
+    * Use formatting from config settings.
+    * Log to screen if ``verbose``
+    * Add a rotating logfile from config settings.
+
+    Args:
+        config (dict): the running config
+        log_name (str, optional): the name of the Logger to modify.
+            If None, use the top level module ('scriptworker').
+            Defaults to None.
+
+    """
+    log_name = log_name or __name__.split(".")[0]
+    top_level_logger = logging.getLogger(log_name)
+
+    datefmt = config["log_datefmt"]
+    fmt = config["log_fmt"]
+    formatter = logging.Formatter(fmt=fmt, datefmt=datefmt)
+
+    if config.get("verbose"):
+        top_level_logger.setLevel(logging.DEBUG)
+        if len(top_level_logger.handlers) == 0:
+            handler = logging.StreamHandler()
+            handler.setFormatter(formatter)
+            top_level_logger.addHandler(handler)
+    else:
+        top_level_logger.setLevel(logging.INFO)
+
+    # Rotating log file
+    os.makedirs(config["log_dir"], exist_ok=True)
+    path = os.path.join(config["log_dir"], file_name)
+    if config["watch_log_file"]:
+        # If we rotate the log file via logrotate.d, let's watch the file
+        # so we can automatically close/reopen on move.
+        handler = logging.handlers.WatchedFileHandler(path)
+    else:
+        # Avoid using WatchedFileHandler during scriptworker unittests
+        handler = logging.FileHandler(path)
+    handler.setFormatter(formatter)
+    top_level_logger.addHandler(handler)
+    top_level_logger.addHandler(logging.NullHandler())
+
+
+# read_config {{{1
+def read_config(config_path="poller.yaml"):
```

Comment:
Hm, possibly. The default isn't good, but we could remove the default or rename it. We could also pass in the default config instead of hardcoding DEFAULT_CONFIG. I might try using `_init_config()` here.

[EDIT] yup, I did it.